### ISSUE TITLE
test_reduce segfaults with MPICH 3.4.1

### ISSUE THREAD

**Issue by teonnik** [eth-cscs/DLA-Future#328]
```
14: Test command: /home/teonnik/code/spack/opt/spack/linux-archrolling-skylake/clang-11.0.1/mpich-3.4.1-xj3lbsaikb7swvbjhgjoqzduhws5gj3z/bin/mpiexec "-n" "4" "/home/teonnik/build/dlaf/aoa3fav/nbmpi/test/unit/communication/test_reduce"
14: Test timeout computed to be: 1500
14: [==========] Running 7 tests from 1 test suite.
14: [----------] Global test environment set-up.
14: [----------] 7 tests from ReduceTest
14: [ RUN      ] ReduceTest.ValueOnSingleRank
14: [       OK ] ReduceTest.ValueOnSingleRank (1 ms)
14: [ RUN      ] ReduceTest.CArrayOnSingleRank
14: [       OK ] ReduceTest.CArrayOnSingleRank (0 ms)
14: [ RUN      ] ReduceTest.Value
14: [       OK ] ReduceTest.Value (1 ms)
14: [ RUN      ] ReduceTest.CArray
14:
14: ===================================================================================
14: =   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
14: =   PID 957933 RUNNING AT T480s
14: =   EXIT CODE: 139
14: =   CLEANING UP REMAINING PROCESSES
14: =   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
14: ===================================================================================
14: YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Segmentation fault (signal 11)
14: This typically refers to a problem with your application.
14: Please see the FAQ page for debugging suggestions
1/1 Test #14: test_reduce ......................***Failed    3.30 sec

0% tests passed, 1 tests failed out of 1

Label Time Summary:
RANK_4    =   3.30 sec*proc (1 test)

Total Test time (real) =   3.31 sec

The following tests FAILED:
	 14 - test_reduce (Failed)
Errors while running CTest
```

**Comment by rasolca** [comment]
Did you install it with spack?
I noticed that the new spack package changed some defaults:
With the new values `device=ch4 netmod=ofi` mpi doesn't even inititalize on macos (crashes during MPI_Init), while if I use the old defaults `device=ch3 netmod=tcp` even 3.4.1 runs correctly
Which settings are you using?

**Comment by teonnik** [comment]
Yes, I installed it with spack with the default values for version 3.4.1 (i.e. device=ch4). I have not tried 3.4.1 with device=ch3 yet.

**Comment by haampie** [comment]
@rasolca can you make an issue about that? seems useful to have a working mpi on macos :p

**Comment by rasolca** [comment]
@haampie I'm not sure what is going on and what is failing (wrong spack package?, mpich bug?, misconfigured dependencies (i.e. spack packages of the dependencies)?). I need to investigate a little bit more.
What I currently know:
- defaults has been changed (even for the old versions) to use ch4 and ofi (libfabric)
- It fails during MPI_Init for nranks >= 2. I suppose (from the error message I get) it is a libfabric problem. 
- I tried to build ch4 with ucx, but it stopped as some dependencies are not available for OSX. I need some time to check the dependency tree to find if I can replace or deactivate these libs.
